### ISSUE TITLE
Incorrect parsing of URLs between parentheses

### ISSUE THREAD

**Issue by joaoubaldo** [signalapp/Signal-Android#9449]
### Bug description
When opening a link between parentheses, the last parenthesis is parsed as being part of the URL. Example: `(https://github.com/signalapp/Signal-Android)`

### Steps to reproduce
- send a message containing the text `(https://github.com/signalapp/Signal-Android)`
- click on that message's link

**Actual result:** the browser opens the url `https://github.com/signalapp/Signal-Android)`
**Expected result:** the browser should open the url `https://github.com/signalapp/Signal-Android`

### Device info
**Device:** Xiaomi Mi A1
**Android version:** LineageOS 16.0
**Signal version:** 4.55.8

**Comment by edent** [comment]
This appears to be an issue with the way Android's built in ` WEB_URL` pattern works.

I've raised it as an issue on https://issuetracker.google.com/issues/161727315

I'm not affiliated with Signal nor Android.

**Comment by edent** [comment]
This is still relevant, but blocked by Android's buggy URL parser.

**Comment by edent** [comment]
I'll raise this againbwith Google. If you have any Android contacts, please ask thrm to look at https://issuetracker.google.com/issues/161727315

**Comment by zenmonkeykstop** [comment]
Just encountered this in the context of Signal Desktop *correctly* parsing a link enclosed in parentheses - so a link that looked to be working for a Desktop user was broken for an Android one. In theory this could have some security impact - you could use it to deliver different content to people based on their client (altho you could probably do that anyway based on user agents).

**Comment by jeffrey-signal** [comment]
> This appears to be an issue with the way Android's built in  WEB_URL pattern works.
> I've raised it as an issue on https://issuetracker.google.com/issues/161727315

Correct, this is caused by [a bug in `Linkify.WEB_URLS`](https://issuetracker.google.com/issues/161727315). We don't currently plan on adding a workaround, so this will need to be fixed by Google.
